relayer/relay.go

```diff
@@ -123,7 +130,19 @@ func (r *Relayer) StartChainListeners(ctx context.Context, errCh chan error) {
 		chainRuntime := chainRuntime
 
 		eg.Go(func() error {
-			return chainRuntime.Provider.Listener(ctx, chainRuntime.LastSavedHeight, chainRuntime.listenerChan)
+			var lastProcessedTxInfo interface{}
+			if chainRuntime.Provider.Type() == "sui" {
```

Comment:
Instead of chain name, can we add some type parameter that will signify this behavioral change for the chain.

Reply:
Actually, we have to implement `lastProcessedTxStore` in all the chains. So, the condition we have here currently will eventually be removed.

Reply:
But most of the chains may never have actual implementation for the last processed Tx or similar logic, so how will it be handled then ?